tests/federation_room_join_partial_state_test.go

```diff
@@ -188,6 +188,249 @@ func TestPartialStateJoin(t *testing.T) {
 	})
 
+	// we should be able to receive typing EDU over federation during the resync
+	t.Run("CanReceiveTypingDuringPartialStateJoin", func(t *testing.T) {
+		deployment := Deploy(t, b.BlueprintAlice)
+		defer deployment.Destroy(t)
+		alice := deployment.Client(t, "hs1", "@alice:hs1")
+
+		server := createTestServer(t, deployment)
+		cancel := server.Listen()
+		defer cancel()
+		serverRoom := createTestRoom(t, server, alice.GetDefaultRoomVersion(t))
+		psjResult := beginPartialStateJoin(t, server, serverRoom, alice)
+		defer psjResult.Destroy()
+
+		derekUserId := psjResult.Server.UserID("derek")
+
+		content, _ := json.Marshal(map[string]interface{}{
+			"room_id": serverRoom.RoomID,
+			"user_id": derekUserId,
+			"typing":  true,
+		})
+		edu := gomatrixserverlib.EDU{
+			Type:    "m.typing",
+			Content: content,
+		}
+		psjResult.Server.MustSendTransaction(t, deployment, "hs1", []json.RawMessage{}, []gomatrixserverlib.EDU{edu})
+
+		psjResult.FinishStateRequest()
+		alice.MustSyncUntil(t,
+			client.SyncReq{},
+			client.SyncEphemeralHas(serverRoom.RoomID, func(result gjson.Result) bool {
+				if result.Get("type").Str != "m.typing" {
+					return false
+				}
+				for _, item := range result.Get("content").Get("user_ids").Array() {
+					if item.Str == derekUserId {
+						return true
+					}
+				}
+				return false
+			}),
+		)
+	})
+
+	// we should be able to receive presence EDU over federation during the resync
+	t.Run("CanReceivePresenceDuringPartialStateJoin", func(t *testing.T) {
+		deployment := Deploy(t, b.BlueprintAlice)
+		defer deployment.Destroy(t)
+		alice := deployment.Client(t, "hs1", "@alice:hs1")
+
+		server := createTestServer(t, deployment)
+		cancel := server.Listen()
+		defer cancel()
+		serverRoom := createTestRoom(t, server, alice.GetDefaultRoomVersion(t))
+		psjResult := beginPartialStateJoin(t, server, serverRoom, alice)
+		defer psjResult.Destroy()
+
+		derekUserId := psjResult.Server.UserID("derek")
+
+		content, _ := json.Marshal(map[string]interface{}{
+			"push": []map[string]interface{}{
+				map[string]interface{}{
+					"user_id":         derekUserId,
+					"presence":        "online",
+					"last_active_ago": 100,
+				},
+			},
+		})
+		edu := gomatrixserverlib.EDU{
+			Type:    "m.presence",
+			Content: content,
+		}
+		psjResult.Server.MustSendTransaction(t, deployment, "hs1", []json.RawMessage{}, []gomatrixserverlib.EDU{edu})
+
+		// for now we just check that the fed transaction is accepted,
+		// presence is not always going down the sync for some reason
+
+		// psjResult.FinishStateRequest()
+		// alice.MustSyncUntil(t,
+		// 	client.SyncReq{},
+		// 	func(userID string, sync gjson.Result) error {
+		// 		for _, e := range sync.Get("presence").Get("events").Array() {
+		// 			t.Logf("lllll %s", e)
+		// 			if e.Get("sender").Str == derekUserId {
+		// 				return nil
+		// 			}
+		// 		}
+		// 		return fmt.Errorf("No presence update from %s", derekUserId)
+		// 	},
+		// )
```

Comment:
Do we suspect a Synapse bug here? If so, is this related to partial joins or a known defect with presence?

Reply:
I am not sure. I would suspect a bug since I vaguely remember seeing both results which is suspicious.
I am planning to have a look at this one, but haven't had the time yet. There is some tracking about my tests&fixes status for EDUs in https://github.com/matrix-org/synapse/issues/13684.